src/components/PeopleTable.tsx

```diff
@@ -1,5 +1,64 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-export const PeopleTable = () => {
+import { Person } from '../types/Person';
+import { Link, useParams, useSearchParams } from 'react-router-dom';
+import { useEffect, useState, useCallback } from 'react';
+import { SearchLink } from '../components/SearchLink';
+import cn from 'classnames';
+
+interface PeopleTableProps {
+  persons: Person[];
+  sortField: string | null;
+  sortOrder: 'asc' | 'desc' | null;
+}
+
+export const PeopleTable: React.FC<PeopleTableProps> = ({
+  persons,
+  sortField,
+  sortOrder,
+}) => {
+  const findPersonByName = (name: string | null) => {
+    return persons.find(p => p.name === name);
+  };
+
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [searchParams] = useSearchParams();
+  const { slug } = useParams();
+
+  useEffect(() => {
+    const storedSlug = searchParams.get('selectedPerson');
```

Comment:
The parameter 'selectedPerson' should be consistent with the key used in the URL search parameters. Ensure that 'selectedPerson' is the correct key or update it to match the actual key used in the URL.